### ISSUE TITLE
[module:download] some enhancementes when using `--composer` needed

### ISSUE THREAD

**Issue by novia713** [hechoendrupal/drupal-console#3172]
### Problem/Motivation
 - Drupal Packagist is deprecated as composer repo
 - At least, it should check if Drupal Packagist is already setted before setting it by default

### Solution
we should use d.o composer repo

**Comment by twhiston** [comment]
I would suggest that this is better dealt with via parsing the project composer file and using anything listed in the repositories section as a potential download source rather than hard coding a url into the command, as it is currently.
This would allow some significant advantages as you could require a module from a private location via drupal console as well.
If @jmolivas agrees with this approach i'm happy to take on the work to realize it

**Comment by novia713** [comment]
yeah, i meant to _add to already existing repos_ the d.o composer repo, given Composer fully supports several repos

**Comment by nvaken** [comment]
Not entirely sure, but if I understand correctly, this issue addresses the problem that `--composer` will add the deprecated Composer repository Drupal Packagist? It should be using the official Drupal Packages server from Drupal.org, right? If so, I'm all in for this one.

**Comment by novia713** [comment]
_apart from me, i guess there isn't already these days anybody that still uses DC instead of Composer to install modules_

so, if it still worths the effort, @jmolivas i would change in `DrupalApi.php`
from

```php
return $this->getComposerReleases(
            sprintf(
                'http://packages.drupal.org/packages/drupal/%s.json',
                trim($module)
            ),
            $limit,
            $unstable
        );
```

to 

```php
return $this->getComposerReleases(
            sprintf(
                'https://packagist.org/packages/drupal/%.json',
                trim($module)
            ),
            $limit,
            $unstable
        );
```

i'm ready to push a PR if it worths
